website/docs/advanced/telemetry.md

```diff
@@ -38,25 +40,167 @@ Here is the call graph of the trace:
+The instrumentation in the TaskWeaver code will send the traces and metrics to the OpenTelemetry collector. 
+An OpenTelemetry collector is a component that receives traces and metrics from the instrumentation, does some processing, and exports them to
+another collector or a backend. In our case, we configure the collector to export the traces to a [Jaeger](https://www.jaegertracing.io/) backend and the metrics 
+to a [Prometheus](https://prometheus.io/) backend.
+![Tracing Architecture](../../static/img/tracing-arch.png)
+
+Both Jaeger and Prometheus are popular open-source monitoring systems. We have prepared a docker-compose file to set up the infrastructure
+in `/TaskWeaver/tracing_configure/docker-compose.yaml`. 
+The content of the file is as follows:
+```yaml
+version: '3'
+services:
+  optl-collector:
+    image: otel/opentelemetry-collector:0.96.0
+    command: ["--config=/etc/collector-config.yaml"]
+    volumes:
+      - ./collector-config.yaml:/etc/collector-config.yaml
+    ports:
+      - "4317:4317" # Expose the gRPC receiver port for the collector
+    depends_on:
+      - jaeger
+
+  jaeger:
+    image: jaegertracing/all-in-one:1.54
+    ports:
+      - "16686:16686" # Jaeger UI
+
+  prometheus:
+    image: prom/prometheus:latest
+    ports:
+      - "9090:9090" # Prometheus UI
+    volumes:
+      - ./prometheus-config.yml:/etc/prometheus/prometheus.yml
+    command: ["--config.file=/etc/prometheus/prometheus.yml"]
+    depends_on:
+      - optl-collector
+```
+If you read the file, you can see that we use the `otl/opentelemetry-collector` image to set up the OpenTelemetry collector,
+We only expose the gRPC receiver port for the collector, which is `4317`.
+The collector configuration file is `collector-config.yaml`, which is mounted to the container.
+The configuration file is as follows:
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: 0.0.0.0:4317
+      http:
+        endpoint: 0.0.0.0:4318
+
+
+exporters:
+  debug:
+    verbosity: detailed
+  otlp:
+    endpoint: "jaeger:4317"
+    tls:
+      insecure: true
+  prometheus:
+    endpoint: "0.0.0.0:9464"
+
+service:
+  pipelines:
+    traces:
+      receivers: [otlp]
+      exporters: [otlp]
+    metrics:
+      receivers: [otlp]
+      exporters: [prometheus]
+    logs:
+      receivers: [otlp]
+      exporters: [debug]
+```
+Because Jaeger is compatible with the OpenTelemetry collector, we can export the traces to Jaeger by setting the `otlp` exporter.
+We also export the metrics to Prometheus by setting the `prometheus` exporter. 
+The `prometheus-config.yml` file is the configuration file for Prometheus, which is as follows:
+```yaml
+scrape_configs:
+  - job_name: optl-collector
+    scrape_interval: 5s
+    static_configs:
+      - targets: ["optl-collector:9464"]
+```
+We only scrape the metrics from the OpenTelemetry collector.
+
+With all the files in place, you can run the following command to set up the infrastructure:
+```bash
```

Comment:
Feel like we can move this up as developers do not need to understand how it was implemented in the beginning.